errorlint/lint.go

```diff
@@ -880,3 +1018,235 @@ func generateErrorVarName(originalName, typeName string) string {
 	// If we couldn't determine a good name, use default.
 	return "anErr"
 }
+
+func resolveConflicts(lints []analysis.Diagnostic, extInfo *TypesInfoExt) []analysis.Diagnostic {
+	ifGroups, otherLints := groupDiagnosticsByIfStmt(lints, extInfo)
+
+	var result []analysis.Diagnostic
+
+	for ifStmt, groupLints := range ifGroups {
+		if len(groupLints) <= 1 {
+			result = append(result, groupLints...)
+			continue
+		}
+
+		if hasConflictingDiagnostics(groupLints) {
+			if combined := createCombinedDiagnostic(ifStmt, groupLints, extInfo); combined != nil {
+				result = append(result, *combined)
+				continue
+			}
+		}
+
+		result = append(result, groupLints...)
+	}
+
+	return append(result, otherLints...)
+}
+
+func findNodeAtPosition(extInfo *TypesInfoExt, pos token.Pos) ast.Node {
+	// First check type-checked expressions (most common case)
+	for node := range extInfo.TypesInfo.Types {
+		if nodeContainsPos(node, pos) {
+			return node
+		}
+	}
+
+	// Fallback: check scopes
+	for scope := range extInfo.TypesInfo.Scopes {
+		if nodeContainsPos(scope, pos) {
+			return scope
+		}
+	}
+
+	return nil
+}
+
+// nodeContainsPos checks if a node contains the given position
+func nodeContainsPos(node ast.Node, pos token.Pos) bool {
+	return node.Pos() <= pos && pos < node.End()
+}
+
+// containingIf finds the if statement that contains the given node
+// by walking up the AST parent chain.
+func containingIf(extInfo *TypesInfoExt, node ast.Node) *ast.IfStmt {
+	current := node
+	for current != nil {
+		if ifStmt, ok := current.(*ast.IfStmt); ok {
+			return ifStmt
+		}
+		parent := extInfo.NodeParent[current]
+		if parent == nil {
+			break
+		}
+		current = parent
+	}
+	return nil
+}
```

Comment:
Maybe this to be consistent

```suggestion
// nodeContainsPos checks if a node contains the given position
func nodeContainsPos(node ast.Node, pos token.Pos) bool {
	return node.Pos() <= pos && pos < node.End()
}

// statementContainsIf finds the if statement that contains the given node
// by walking up the AST parent chain.
func statementContainsIf(extInfo *TypesInfoExt, node ast.Node) *ast.IfStmt {
	current := node
	for current != nil {
		if ifStmt, ok := current.(*ast.IfStmt); ok {
			return ifStmt
		}
		parent := extInfo.NodeParent[current]
		if parent == nil {
			break
		}
		current = parent
	}
	return nil
}
```